haystack/preview/components/file_converters/txt.py

```diff
@@ -21,189 +15,45 @@ class TextFileToDocument:
-        progress_bar: Optional[bool] = None,
-    ):
+    def run(self, sources: List[Union[str, Path, ByteStream]]):
         """
         Convert text files to Documents.
 
-        :param paths: A list of paths to text files.
-        :param metadata: Optional metadata to attach to the Documents. If a list is provided, the length of the list
-            must match the number of paths. Default: `None`
-        :param encoding: The encoding of the text files. Default: `"utf-8"`
-        :param remove_numeric_tables: This option uses heuristics to remove numeric rows from the tables.
-            The tabular structures in documents might be noise for reader models if they don't have table parsing
-            capability for finding answers. However, tables may also have long strings that could be possible candidates
-            for answers. The rows containing strings are thus retained in this option. Default: `False`
-        :param numeric_row_threshold: Applicable if `remove_numeric_tables` is set to `True`. This is the threshold to
-            determine if a line in the provided text file is a numeric table row or not. The value is the ratio of
-            numeric words to the total number of words in a line. Default: `0.4`
-        :param valid_languages: Validate languages from a list of languages specified in the
-            [ISO 639-1 format]((https://en.wikipedia.org/wiki/ISO_639-1)). This option can be used to add a test for
-            encoding errors. If the extracted text is not one of the valid languages, then there might be an encoding
-            error resulting in garbled text. Default: `None`
-        :param progress_bar: Whether to show a progress bar for the conversion process. Default: `True`
+        :param streams: A list of paths to text files or ByteStream objects.
+            Note that if an encoding is specified in the metadata of a ByteStream,
+            it will override the component's default.
+        :return: A dictionary containing the converted documents.
         """
-        if encoding is None:
-            encoding = self.encoding
-        if remove_numeric_tables is None:
-            remove_numeric_tables = self.remove_numeric_tables
-        if numeric_row_threshold is None:
-            numeric_row_threshold = self.numeric_row_threshold
-        if valid_languages is None:
-            valid_languages = self.valid_languages
-        if progress_bar is None:
-            progress_bar = self.progress_bar
-
-        metas = TextFileToDocument._prepare_metadata(metadata, paths)
-
         documents = []
-        for path, meta in tqdm(
-            zip(paths, metas), total=len(paths), desc="Converting text files", disable=not progress_bar
-        ):
+        for source in sources:
+            if isinstance(source, (Path, str)):
+                if not Path(source).exists():
+                    logger.warning("File %s does not exist. Skipping it.", source)
+                    continue
```

Comment:
Why mixing up EAFP and LBYL? Let's remove this block, if the file doesn't exist we'll get a meaningful error at line 49